### ISSUE TITLE
Specify FLAG UTF-8 when converting to UTF-8, if there was no explicit FLAG option

### ISSUE THREAD

**Issue by donnerpeter** [wooorm/dictionaries#25]
Hunspell read the affix file byte by byte and decodes UTF-8 on demand. If it's not instructed to do so for flags, it doesn't. So non-ASCII characters like "ý" are treated like several characters, and due to another [bug](https://github.com/hunspell/hunspell/issues/707) Hunspell silently takes just the first character and ignores the rest. So the words can have unexpected flags.

Example: `pt` contains `FORBIDDENWORD ý`, and the perfectly valid word `trabalhar/akYMjLÀÚ` is treated as having this flag and thus considered misspelled.

**Comment by wooorm** [comment]
Yeah good idea. I do remember thinking about this, but it never came up. Perhaps a send expression in crawl.sh could do the trick. PR welcome!

**Comment by donnerpeter** [comment]
Yes, some combination of bash and unix text processing utilities should help. Neither of them are my strong side, so I wouldn't hold breath from a PR by me in the very near future :)

**Comment by wooorm** [comment]
Shouldn’t this issue be about setting an `SET UTF-8` instead of using a `FLAG UTF-8`? 🤔

**Comment by donnerpeter** [comment]
No, it's not enough. At the moment of submisson `pt` already had `SET UTF-8`, but Hunspell parses flags byte by byte, and needs to know that they're in UTF-8, too.

**Comment by wooorm** [comment]
That sounds complex...

But, then this is a bug in Portuguese though? *It* should either use ASCII flags, or `SET UTF-8`?

**Comment by wooorm** [comment]
Hmm, that still seems like an issue for them though? That should be fixed in the upstream, rather than patched here?

**Comment by donnerpeter** [comment]
The issue should be addressed where the dictionaries are converted into UTF-8. My understanding was that it was here, at least partly. If I'm mistaken, then this is a wrong repo indeed :)
